src/sqlfluff/dialects/dialect_redshift.py

```diff
@@ -377,3 +379,127 @@ class CreateGroupSegment(BaseSegment):
+
+    https://docs.aws.amazon.com/redshift/latest/dg/r_ALTER_USER.html
+    """
+
+    type = "alter_user"
+
+    match_grammar = Sequence(
+        "ALTER",
+        "USER",
+        Ref("NakedIdentifierSegment"),
+        Ref.keyword("WITH", optional=True),
+        AnyNumberOf(
+            OneOf(
+                "CREATEDB",
+                "NOCREATEDB",
+            ),
+            OneOf(
+                "CREATEUSER",
+                "NOCREATEUSER",
+            ),
+            Sequence(
+                "SYSLOG",
+                "ACCESS",
+                OneOf(
+                    "RESTRICTED",
+                    "UNRESTRICTED",
+                ),
+            ),
+            Sequence(
+                "PASSWORD",
+                OneOf(
+                    Ref("QuotedLiteralSegment"),
+                    "DISABLE",
+                ),
+                Sequence("VALID", "UNTIL", Ref("QuotedLiteralSegment"), optional=True),
+            ),
+            Sequence(
+                "RENAME",
+                "TO",
+                Ref("NakedIdentifierSegment"),
+            ),
+            Sequence(
+                "CONNECTION",
+                "LIMIT",
+                OneOf(
+                    Ref("NumericLiteralSegment"),
+                    "UNLIMITED",
+                ),
+            ),
+            OneOf(
+                Sequence(
+                    "SESSION",
+                    "TIMEOUT",
+                    Ref("NumericLiteralSegment"),
+                ),
+                Sequence(
+                    "RESET",
+                    "SESSION",
+                    "TIMEOUT",
+                ),
+            ),
+            OneOf(
+                Sequence(
+                    "SET",
+                    Ref("NakedIdentifierSegment"),
+                    OneOf(
+                        "TO",
+                        Ref("EqualsSegment"),
+                    ),
+                    OneOf(
+                        "DEFAULT",
+                        Ref("LiteralGrammar"),
+                    ),
+                ),
+                Sequence(
+                    "RESET",
+                    Ref("NakedIdentifierSegment"),
+                ),
+            ),
+            min_times=1,
+        ),
+    )
+
+
+@redshift_dialect.segment()
+class AlterGroupSegment(BaseSegment):
+    """`ALTER GROUP` statement.
+
+    https://docs.aws.amazon.com/redshift/latest/dg/r_ALTER_GROUP.html
+    """
+
+    type = "alter_group"
+
+    match_grammar = Sequence(
+        "ALTER",
+        "GROUP",
+        Ref("NakedIdentifierSegment"),
+        OneOf(
+            Sequence(
+                "ADD",
+                "USER",
+                Delimited(
+                    Ref("NakedIdentifierSegment"),
+                ),
+            ),
+            Sequence(
+                "DROP",
+                "USER",
+                Delimited(
+                    Ref("NakedIdentifierSegment"),
+                ),
+            ),
```

Comment:
Could merge these if you wanted:

```suggestion
            Sequence(
                OneOf("ADD", "DROP"),
                "USER",
                Delimited(
                    Ref("NakedIdentifierSegment"),
                ),
            ),
```